src/cpl/nuopc/lnd_import_export.F90

```diff
@@ -295,11 +297,19 @@ subroutine advertise_fields(gcomp, flds_scalar_name, glc_present, cism_evolve, r
     end if
 
     ! export to glc
-    if (glc_present .and. cism_evolve) then
-       ! lnd->glc states from land all lnd->glc elevation classes (1:glc_nec) plus bare land (index 0).
-       ! The following puts all of the elevation class fields as an
-       ! undidstributed dimension in the export state field
+    call NUOPC_CompAttributeGet(gcomp, name="histaux_l2x1yrg", value=cvalue, &
+         isPresent=isPresent, isSet=isSet, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    if (isPresent .and. isSet) then
+       read(cvalue,*) send_lnd2glc
+    else
+       send_lnd2glc = .false.
+    end if
 
+    if ((send_lnd2glc) .or. (glc_present .and. cism_evolve)) then
```

Comment:
- [ ] I find it confusing that send_lnd2glc is set entirely based on the value of histaux_l2x1yrg. That means that, if histaux_l2x1yrg is false, then send_lnd2glc will be false even if you're running with an active glc model. This seems error-prone if someone introduces more logic later based on send_lnd2glc. Would it work to have logic like:
  - if (glc_present .and. cism_evolve) then send_lnd2glc = .true.
  - else set lnd_2glc based on the new logic based on histaux_l2g1yrg

- [ ] Also, I see that lnd -> glc fields aren't sent if cism_evolve is false. I can't remember if we discussed this before, but this strikes me as wrong: it seems like lnd2glc fields should be sent whenever glc_present is true, because land ice folks will sometimes set up a non-evolving ice sheet just to get the downscaled lnd2glc fields. Or is there some reason why this won't work right now with cmeps?

Reply:
@sacks - good points. I think the right thing to do would be the following:
- if (glc_present) then send_lnd2glc = .true.
- else (glc_present .and. cism_evolve)
I think this takes care of the two concerns you had. Do you agree? In this case if glc_present you will always send the lnd2glc fields otherwise you simply queyr the histaux_l2g1yrg

Reply:
I'm not sure what you mean by the condition on the else (`else (glc_present .and. cism_evolve)`): is that a mistake? From your description in the last sentence, I agree.